app/src/main/java/com/montreal/wtm/ui/fragment/ProgramFragment.java

```diff
@@ -143,32 +134,57 @@ public String getStartDate(String stringDate, String startTime, String endTime,
             long timeEnd = dateEnd.getTime();
             Log.v("Date", "Date = " + dateStart);
 
-            double difference = Math.floor((timeEnd - timeStart) / totalNumber);
-            Log.v("Date", "difference = " + difference);
+            long difference = (timeEnd - timeStart) / totalNumber;
+            Log.v("Date", "difference = " + new Date(difference));
 
-            long newDateTime = (long) (timeStart + difference * number );
+            long newStartTime = timeStart + difference * number;
+            long newEndTime = timeStart + difference * (number + 1);
 
-            Date result = new Date(newDateTime);
-            int minutes = result.getMinutes();
-            minutes = minutes > 9 ? minutes : '0' + minutes;
-            Log.v("result", " result = " + result.getHours() + ":" + minutes);
-            return result.getHours() + ":" + minutes;
+            return getTime(newStartTime) + " - " + getTime(newEndTime);
         } catch (ParseException e) {
-            e.printStackTrace();
+            Log.e(TAG, "ParseException", e);
         }
         return startTime + " " + endTime;
     }
 
+    protected String getTime(long timeInMillis) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeInMillis(timeInMillis);
+        StringBuilder builder = new StringBuilder();
+        builder.append(calendar.get(Calendar.HOUR_OF_DAY));
+        builder.append(":");
+        int minutes = calendar.get(Calendar.MINUTE);
+        if (minutes <= 9) {
+            builder.append("0");
+        }
+        builder.append(minutes);
+        return builder.toString();
+    }
+
     @Override
     public void retryOnProblem() {
         if (!isAdded()) {
             return;
         }
-        FirebaseData.INSTANCE.getSessions(getActivity(), requestListenerSession);
 
         FirebaseData.INSTANCE.getMyShedule(getActivity(), requestListenerMySchedule);
+        FirebaseData.INSTANCE.getSessions(getActivity(), requestListenerSession);
+        FirebaseData.INSTANCE.getMyRatings(getActivity(), requestListenerRatings);
     }
 
+    private FirebaseData.RequestListener<HashMap<String, Long>> requestListenerRatings =
+        new FirebaseData.RequestListener<HashMap<String, Long>>() {
+            @Override
+            public void onDataChange(@org.jetbrains.annotations.Nullable HashMap<String, Long> data) {
+                Log.d(TAG, "Received ratings:" + data);
```

Comment:
remove log

Reply:
Won't remove the logs, they should be omitted with proguard

Reply:
Ok